untitled/main/com/aca/homework/week11/occurence/KeyCalculator.java

```diff
@@ -0,0 +1,32 @@
+package com.aca.homework.week11.occurence;
+
+import java.util.*;
+
+public class KeyCalculator {
+    private Supplier supplier;
```

Comment:
this is a raw usage of generic interface. please do not do that. That's why you have to cast to String in some places. Instead, you can use Supplier<String>

Reply:
could be final because once you set it and never change